CodePush.m

```diff
@@ -237,6 +285,22 @@ - (void)loadBundle
     });
 }
 
+- (void)recordDeploymentStatusReported:(NSString *)appVersionOrPackageIdentifier
+                                status:(NSString *)status
+{
+    NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+    NSMutableDictionary *sentStatusReports = [preferences objectForKey:StatusReportsKey];
+    if (sentStatusReports == nil) {
+        sentStatusReports = [NSMutableDictionary dictionary];
+    } else {
+        sentStatusReports = [sentStatusReports mutableCopy];
+    }
+    
+    [sentStatusReports setValue:status forKey:appVersionOrPackageIdentifier];
```

Comment:
This dictionary will be ever growing, never shrinking?

Can't we clear it at some point, such as before adding a successful deployment status?